virttest/utils_misc.py

```diff
@@ -2818,199 +2818,314 @@ def get_uptime(session=None):
+    :param n_partitions: the number of partitions on disk for linux guest
+    :param force: whether force format partition for windows guest
+    :param mountpoint: mount point list for all partitions in guest.
     :param size: partition size.
     :param fstype: filesystem type for the disk; when it's the default None,
                    it will use the default one for corresponding ostype guest
-    :return Boolean: disk usable or not.
+    :param labeltype: label type for the disk.
+    :return a list: mount point list for all partitions.
     """
     default_fstype = "ntfs" if (ostype == "windows") else "ext3"
     fstype = fstype or default_fstype
     if ostype == "windows":
-        return format_windows_disk(session, did, mountpoint, size, fstype)
-    return format_linux_disk(session, did, all_disks_did, partition,
-                             mountpoint, size, fstype)
+        return format_windows_disk(session, dids, mountpoint, size, fstype, force)
+    value = []
+    all_disks_did = sorted(get_linux_disks(session, partition).values(),
+                           key=lambda x: x[0])
+    for did in dids:
+        p = format_linux_disk(session, did, all_disks_did,
+                              partition, n_partitions,
+                              mountpoint, size, fstype, labeltype)
+        if not p:
+            return []
+        value.extend(p)
+    return value
+
+
+def delete_linux_partition(session, kname):
+    """
+    remove all partitions for one disk.
+
+    :param session: session object to guest.
+    :param kname: disk kname. e.g. sdb
+    """
+    session.cmd_output_safe("partprobe /dev/%s" % kname, timeout=600)
+    list_disk_cmd = "lsblk -o KNAME,MOUNTPOINT"
+    output = session.cmd_output_safe(list_disk_cmd, timeout=600)
+    output = output.splitlines()
+    regex_str = kname + "(\d+)"
+    rm_cmd = "parted -s /dev/%s rm %s"
+    for line in output:
+        partition = re.findall(regex_str, line, re.I | re.M)
+        if partition:
+            if "/" in line.split()[-1]:
+                session.cmd_output_safe("umount %s" % line.split()[-1], timeout=600)
+            logging.info("remove partition %s on %s" % (partition[0], kname))
+            session.cmd_output_safe(rm_cmd % (kname, partition[0]), timeout=600)
+            session.cmd_output_safe("partprobe /dev/%s" % kname, timeout=600)
+
+
+def delete_windows_partition(session, did):
+    """
+    remove all partitions for one disk.
+
+    :param session: session object to guest.
+    :param did: disk index which show in 'diskpart list disk'.
+    """
+    cmd_header = "echo list disk > disk && echo select disk %s >> disk &&"
+    cmd_footer = "&& echo exit >> disk && diskpart /s disk && del /f disk"
+    clean_cmd = "echo clean >> disk"
+    clean_cmd = " ".join([cmd_header, clean_cmd, cmd_footer])
+    session.cmd_output_safe(clean_cmd % did, timeout=600)
+
+
+def delete_guest_partition(session, dids, ostype):
```

Comment:
what about provide a default value for `ostype`

Reply:
I think it is not necessary to add it. And ostype is also used in format_guest_disk. Set the value by user is more clearer when call for the function.